src/components/textarea/textarea.test.tsx

```diff
@@ -17,28 +17,18 @@ jest.mock("../../__internal__/utils/helpers/guid");
 const mockedGuid = "guid-12345";
 (guid as jest.MockedFunction<typeof guid>).mockImplementation(() => mockedGuid);
 
-let loggerSpy: jest.SpyInstance<void, [message: string]> | jest.Mock;
-
-beforeEach(() => {
-  loggerSpy = jest.spyOn(Logger, "deprecate");
-});
-
-afterEach(() => {
-  loggerSpy.mockRestore();
-});
-
-afterAll(() => {
-  loggerSpy.mockClear();
-});
-
 test("should display deprecation warning once when rendered as uncontrolled", () => {
+  const loggerSpy = jest.spyOn(Logger, "deprecate");
+
   render(<Textarea name="my-textarea" defaultValue="test" />);
 
   expect(loggerSpy).toHaveBeenCalledWith(
     "Uncontrolled behaviour in `Textarea` is deprecated and support will soon be removed. Please make sure all your inputs are controlled."
   );
 
   expect(loggerSpy).toHaveBeenCalledTimes(1);
+
+  loggerSpy.mockClear();
```

Comment:
**suggestion**: what do you think about doing this, so the original implementation is restored?

```suggestion
  loggerSpy.mockRestore();
```

Reply:
oh yeah, not sure how/why I ended up with the wrong method here when I think it was correct before I cleaned it up!